api/rest/restcore/projects_rest.php

```diff
@@ -42,8 +42,16 @@
 	$g_app->delete( '/{id}/', 'rest_project_delete' );
 
 	# Project versions
+	$g_app->get( '/{id}/versions', 'rest_project_version_get' );
+	$g_app->get( '/{id}/versions/', 'rest_project_version_get' );
+	$g_app->get( '/{id}/versions/{version_id}', 'rest_project_version_get' );
+	$g_app->get( '/{id}/versions/{version_id}/', 'rest_project_version_get' );
 	$g_app->post( '/{id}/versions', 'rest_project_version_add' );
 	$g_app->post( '/{id}/versions/', 'rest_project_version_add' );
+	$g_app->patch( '/{id}/versions/{version_id}', 'rest_project_version_update' );
+	$g_app->patch( '/{id}/versions/{version_id}/', 'rest_project_version_update' );
+	$g_app->delete( '/{id}/versions/{version_id}', 'rest_project_version_delete' );
+	$g_app->delete( '/{id}/versions/{version_id}/', 'rest_project_version_delete' );	
```

Comment:
Shouldn't `/{id}/versions` be defined as a group to avoid duplication?

Also, what about making use of optional parameters so handle the route variants for a given method ? Something like
```php
$g_app->get( ''/{id}/versions[/{version_id}][/]', 'rest_project_version_get' );
```
(I have not actually tested this)